src/rust/ide/parser/tests/ast.rs

```diff
@@ -48,17 +48,17 @@ pub fn generalized_infix_test() {
 
     let infix = make_gen_infix("a+b").unwrap();
     assert_eq!(infix.name(),"+");
-    assert_eq!(infix.left.repr(),"a");
-    assert_eq!(infix.right.repr(),"b");
+    assert_eq!(infix.left.as_ref().map(|a| &a.arg).repr(),"a");
+    assert_eq!(infix.right.as_ref().map(|a| &a.arg).repr(),"b");
```

Comment:
I cannot implement HasTokens for Operand as it is, because I don't know if the offset is before the argument, or after. I could add a flag for it, but I'm not sure if it is cleaner.

Reply:
Fair enough.